src/pbkdf2.rs

```diff
@@ -150,14 +148,42 @@ pub fn derive(prf: &'static PRF, iterations: usize, salt: &[u8], secret: &[u8],
     // Clear |out|.
     polyfill::slice::fill(out, 0);
 
+    let mut idx: u32 = 0;
+
+    for chunk in out.chunks_mut(output_len) {
+        idx += 1;
+        let chunk_len = chunk.len();
+
+        if chunk_len == output_len {
+            derive_block(&secret, iterations, salt, idx, output_len, chunk);
```

Comment:
Because `derive_block` expects `output_len`-sized blocks, and if `chunk_len` is smaller than that it will cause an out-of-bounds write.